service/debugger/debugger.go

```diff
@@ -420,6 +439,21 @@ func (d *Debugger) Restart(rerecord bool, pos string, resetArgs bool, newArgs []
 	}
 	var p *proc.Target
 	var err error
+
+	if rebuild {
+		// We cannot build processes that we don't know how to build. If
+		// d.config.Kind is different from ExecutingGeneratedFile that means we
+		// didn't build the binary and we are just running it
+		if d.config.Kind == ExecutingExistingFile {
+			err = gobuild.GoTestBuild(d.processArgs[0], d.config.Packages, d.config.BuildFlags)
```

Comment:
This is still wrong:

ExecutingExistingFile, ExecutingOther → it can't be rebuilt at all (either an separately built executable file or a core dump/recording)
ExecutingGeneratedFile → it can be rebuilt by GoBuild
ExecutingGeneratedTest → it can be rebuilt by GoTestBuild

it doesn't make sense to call GoTestBuild on ExecutingExistingFile and it doesn't make sense to call GoBuild on anything other than ExecutingGeneratedFile.

Reply:
Updated. Thanks for the explanation :)